File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0; 
        int right = heights.size()-1;
        int max = 0;
        while(left <right){
            int cur = std::min(heights[right],heights[left])*(right-left);
            if(heights[right] < heights[left]){
                right--;
            }else{
                left++;
            }
            if(cur >max){
                max = cur;
            }
        } 
        return max;


    }
};

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        
    # either looking for both. or found one and looking for other. 
    # bst so not completely random. we know how to partition. 
    # at every point we know the targets. if we have found one and if the other will be left or right
    # also know the current one 
        self.p = p
        self.q = q
        return self.recurs(root)

    def recurs(self, cur):
       # if we havent found any and targets are both less we search left and right, 
       #keep going until different sides

        if p.val <cur.val and q.val < cur.val:
            return self.recurs(cur.left)
        elif p.val >cur.val and q.val > cur.val:
            return self.recurs(cur.right)
        return cur
